File: frontend/src/lib/types/analytics.ts
export type KPIType =
  | "PATIENTS_ACTIVE"
  | "APPOINTMENTS_TODAY"
  | "REVENUE_MONTH"
  | "ATTENDANCE_RATE"
  | "TREATMENT_ACCEPTANCE"
  | "AVG_TICKET"
  | "NEW_PATIENTS_MONTH"
  | "CANCELLATION_RATE";

export interface KPI {
  kpi_type: KPIType;
  value: number;
  label: string;
  formatted_value: string;
  trend: "up" | "down" | "stable" | null;
}

export interface DashboardKPIs {
  kpis: KPI[];
}

export interface ClinicReport {
  tenant_id: string;
  period_start: string;
  period_end: string;
  kpis: KPI[];
  generated_at: string;
}

File: backend/src/odontoflow/api/schemas/analytics.py
"""Pydantic v2 schemas for Analytics endpoints."""
from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class KPIResponse(BaseModel):
    kpi_type: str
    value: float
    label: str
    formatted_value: str
    trend: Optional[str] = None


class DashboardKPIsResponse(BaseModel):
    kpis: list[KPIResponse]


class ClinicReportResponse(BaseModel):
    tenant_id: str
    period_start: str
    period_end: str
    kpis: list[KPIResponse]
    generated_at: str
